engine/src/flutter/display_list/dl_builder.cc

```diff
@@ -1217,6 +1257,30 @@ void DisplayListBuilder::DrawDiffRoundRect(const DlRoundRect& outer,
   SetAttributesFromPaint(paint, DisplayListOpFlags::kDrawDRRectFlags);
   drawDiffRoundRect(outer, inner);
 }
+void DisplayListBuilder::DrawRoundSuperellipse(const DlRoundSuperellipse& rse,
+                                               const DlPaint& paint) {
+  SetAttributesFromPaint(paint, DisplayListOpFlags::kDrawRSuperellipseFlags);
+  drawRoundSuperellipse(rse);
+}
+void DisplayListBuilder::drawRoundSuperellipse(const DlRoundSuperellipse& rse) {
+  if (rse.IsRect()) {
+    drawRect(rse.GetBounds());
+  } else if (rse.IsCircle()) {
```

Comment:
Can oval be optimized? (Thinking) I guess not because the curvature is exact for a circle, but not for an oval?

Reply:
Yes it can! When this PR was drafted RSE didn't support elliptical radii yet, but now yes. Thanks for pointing it out. I almost forgot.

Reply:
So, it could, but we don't support them yet anyway?

Reply:
The code you were commenting was created 2 months ago. I've updated it to support Ovals this evening.

Reply:
Also, considering the complexities you've mentioned I'll probably implement stroke before continuing this one.